src/test/java/com/juniorstart/juniorstart/controller/ProjectControllerTest.java

```diff
@@ -0,0 +1,182 @@
+                .andExpect(jsonPath("$.content.[0].recruiting",is(project.getRecruiting())))
+                .andExpect(jsonPath("$.content.[0].numberOfSeats",is(12)));
+
+    }
+
+    @Test
+    public void getProjectByTitle() throws Exception {
+        mockMvc.perform(get(BASIC_PROJECT_URL + "/title/" + project.getTitle())).andExpect(status().isOk())
+                .andExpect(jsonPath("$.content.[0].name",is(project.getName())))
+                .andExpect(jsonPath("$.content.[0].title",is(project.getTitle())))
+                .andExpect(jsonPath("$.content.[0].description",is(project.getDescription())))
+                .andExpect(jsonPath("$.content.[0].repository",is(project.getRepository())))
+                .andExpect(jsonPath("$.content.[0].recruiting",is(project.getRecruiting())))
+                .andExpect(jsonPath("$.content.[0].numberOfSeats",is(12)));
+
+    }
+
+    @Test
+    public void getProjectByName() throws Exception {
+        mockMvc.perform(get(BASIC_PROJECT_URL + "/name/" + project.getName())).andExpect(status().isOk())
+                .andExpect(jsonPath("$.name",is(project.getName())))
+                .andExpect(jsonPath("$.description",is(project.getDescription())))
+                .andExpect(jsonPath("$.repository",is(project.getRepository())))
+                .andExpect(jsonPath("$.numberOfSeats",is(12)));
+    }
+
+    @Test
+    @WithUserDetails("someEmail")
+    public void createProject() throws Exception {
+
+        String jsonRequest = gson.toJson(projectRequest);
+
+
+        mockMvc.perform(post(BASIC_PROJECT_URL)
+                .contentType(MediaType.APPLICATION_JSON)
+                .characterEncoding("UTF-8")
+                .content(jsonRequest))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.name",is(projectRequest.getName())))
+                .andExpect(jsonPath("$.description",is(projectRequest.getDescription())));
+    }
+
+    @Test
+    @WithUserDetails("someEmail")
+    public void deleteProject() throws Exception {
+        mockMvc.perform(delete(BASIC_PROJECT_URL + "/" + project.getId())).andExpect(status().isOk())
+                .andExpect(jsonPath("success",is(true)))
+                .andExpect(jsonPath("message",is("Project deleted successfully"))).andReturn();
+    }
+
+    @Test
+    @WithUserDetails("someEmail")
+    public void updateProject() throws Exception {
+
+        String jsonRequest = gson.toJson(projectRequest);
+
+        mockMvc.perform(put(BASIC_PROJECT_URL + "/" + project.getId())
+                .contentType(MediaType.APPLICATION_JSON)
+                .characterEncoding("UTF-8")
+                .content(jsonRequest))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.name",is(projectRequest.getName())))
+                .andExpect(jsonPath("$.description",is(projectRequest.getDescription())));
+    }
```

Comment:
Testy napisane czytelnie. Bardzo dobrze to napisałeś, jedynie @DispayName bym jeszcze pododawał.